### PR TITLE
fix for amalgamation build with MIN=1

### PR DESCRIPTION
**Pull request by asmushetzel**
Fixing the first issue listed in PR 7541.

### CODE REVIEW

**Comment by asmushetzel** [comment]
All tests ok except that MKMKL-build failed in one test where it tries to find the best convolution algorithm and ran out of memory on the GPU. Should be unrelated to this change here.

**Comment by piiswrong** [comment]
@mli It feels like some workers have hanging jobs on them that's taking up GPU memory.

File: src/operator/fully_connected-inl.h [hunk 1]
```diff
@@ -108,9 +110,12 @@ class FullyConnectedOp : public Operator {
           Shape2(oshape[0], oshape.ProdShape(1, oshape.ndim())), s);
     }
 
-    // Legacy approach shown here for comparison:
-    //   out = dot(data, wmat.T());
+#if (MSHADOW_USE_CBLAS == 0)
+    // Legacy approach for amalgamation build w/out cblas
+    out = dot(data, wmat.T());
```
Comment (piiswrong) [inline src/operator/fully_connected-inl.h]: I'll merge this for now but eventually we should add a non-blas version into linalg_gemm

Comment (asmushetzel) [inline src/operator/fully_connected-inl.h]: Question is where we draw the line. linalg is all blas/lapack. And we certainly don't want to provide non-blas versions for everything. We can do it with gemm as this is the thing that is used everywhere. 
Seems like we should scope it somehow of what we all want to support also with non-blas implementations. Any suggestions? Should we ask Steffen's team whether they can add this?

**Comment by mli** [comment]
i guess we have set timeout on each jenkins job? before i manually stop hanging jobs, now i don't have the permission to do it anymore

**Comment by DickJC123** [comment]
I was in the process of switching the 21 uses of dot() over to linalg_gemm when I noticed this change.  If the preferred long-term interface is linalg_gemm, why don't I just modify linalg_impl.h to check MSHADOW_USE_CBLAS and invoke dot(), rather than have this ifdef sequence repeated in many places?

File: src/operator/fully_connected-inl.h (hunk 1 above)
Comment (piiswrong) [inline src/operator/fully_connected-inl.h]: We can think of it as a "MXBlas" that only implements gemm.

**Comment by piiswrong** [comment]
@DickJC123 Yeah I think that's reasonable. A better solution would be to copy the dot implementation into mxnet and use it when blas is not available.

**Comment by DickJC123** [comment]
I would prefer to limit the scope of the change I'm working on, so I'll go forward with my suggestion.  I'll leave it to another time/person to copy the dot implementation from mshadow, if that remains desirable.

**Comment by eric-haibin-lin** [comment]
Is the `MIN=1` arg added back to Jenkins now?

**Comment by rahul003** [comment]
Looks like this issue hasn't been fixed yet. 
https://builds.apache.org/blue/organizations/jenkins/incubator-mxnet/detail/master/243/pipeline/50 Please check if something is missed.

**Pull request merged**